setup.gmk

```diff
@@ -65,11 +65,11 @@ else
 TOOLPREFIX=${TOOLCHAIN}/bin/${ARCH}-
 endif
 
-CC=${TOOLPREFIX}gcc
-LD=${TOOLPREFIX}ld
-AR=${TOOLPREFIX}ar
-OD=${TOOLPREFIX}objdump
-OC=${TOOLPREFIX}objcopy
+CC ?= ${TOOLPREFIX}gcc
+LD ?= ${TOOLPREFIX}ld
+AR ?= ${TOOLPREFIX}ar
+OD ?= ${TOOLPREFIX}objdump
+OC ?= ${TOOLPREFIX}objcopy
```

Comment:
Could you please describe what the intended outcome of this change is?

Reply:
The current makefile gives some constraints:

1. The toolchain must be in a `/toolchain-${platform} folder`
2. Toolchain binaries must reside in a `bin` folder
3. All toolchain binaries must reside in the same folder

Constraints 2-3 are generally met by most environments. Constraint 1 is problematic. E.g. in yocto, 1 is not met, 2.3 are.

- The proposed change: `TOOLCHAIN?=${HOME}/toolchain-${platform}` removes constraint 1.
- The one you linked removes constraint 2-3.

In my case (yocto), removing constraint 1 is mandatory.

Reply:
I just noticed that yocto does not produce a tree compatible with constraint 2. So both changes are necessary.

Also, note that yocto will by default set CC, LD, ... vars in the build setup process.

Reply:
This change is a problem because GNU make populates CC and AR with default values. Hence `?=` has no effect for those variables unless make is called with `-R` flag. That's why we cannot simply accept with change. 
Thanks for the description of your use case. We will take this into account for future releases.